dialoptions.go

```diff
@@ -446,6 +448,22 @@ func WithDisableServiceConfig() DialOption {
 	})
 }
 
+// WithDefaultServiceConfig returns a DialOption that configures the default service config, which
+// will be used in cases where:
+// 1. WithDisableServiceConfig is called.
+// 2. Resolver does not return service config or if the resolver gets and invalid config.
+func WithDefaultServiceConfig(s string) DialOption {
+	return newFuncDialOption(func(o *dialOptions) {
+		sc, err := parseServiceConfig(s)
```

Comment:
We still give users a chance to pass invalid string here.

How about returning a dial option and an error from `ValidateServiceConfig()`, so the only way to create a default service config dial option is to validate it first?

The name `ValidateServiceConfig` would be inappropriate, time to bikeshed.

Reply:
If we have something like
```
func WithDefaultServiceConfig(s string) (DialOption, error) {
	sc, err := parseServiceConfig(s)
	if err != nil {
		return nil, fmt.Errorf("the provided service config is invalid, err: %v", err)
	}
	return newFuncDialOption(func(o *dialOptions) {
		o.defaultServiceConfig = sc
	}), nil
}
```
Then it breaks the idiomatic way of doing `grpc.Dial(target, grpc.WithXXX(), grpc.WithYYY())`. I think separating the validation is ok. I can add a comment saying that it's strongly recommended to verify the validity of the json string with the `ValidateServiceConfig` function.

Reply:
Per offline discussion, we will check the validity of the default service config in `DialContext`. If it is not valid, the Dial will fail.